From 16907b2942fc12f4630e56a60943d752bee28dae Mon Sep 17 00:00:00 2001
From: Kamil Debski <k.debski@samsung.com>
Date: Wed, 22 May 2013 15:46:21 +0200
Subject: [PATCH 0318/1302] modem_if: device_tree: Add modem_if to
 exynos4412-slp_pq.dts

Add modem_if to the device tree of slp_pq board.

Signed-off-by: Kamil Debski <k.debski@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 arch/arm/boot/dts/exynos4412-slp_pq.dts | 42 +++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/arch/arm/boot/dts/exynos4412-slp_pq.dts b/arch/arm/boot/dts/exynos4412-slp_pq.dts
index a9df9c6..4695393 100644
--- a/arch/arm/boot/dts/exynos4412-slp_pq.dts
+++ b/arch/arm/boot/dts/exynos4412-slp_pq.dts
@@ -993,4 +993,46 @@
 		vdd_arm-supply = <&buck2_reg>;
 		status = "okay";
 	};
+
+	modem_if {
+		compatible = "samsung,modem_if";
+		reset-req-gpio = <&gpm3 3 0>;
+		cp-on-gpio = <&gpl2 5 0>;
+		cp-reset-gpio = <&gpx3 2 0>;
+		pda-active-gpio = <&gpf1 6 0>;
+
+		phone-active-gpio = <&gpx1 6 0>;
+		cp-dump-int-gpio = <&gpx1 2 0>;
+
+		ap-dump-int-gpio = <0>;
+		flm_uart_sel = <0>;
+		cp_warm_reset = <0>;
+		sim_detect = <0>;
+
+		link-slavewake-gpio = <&gpx1 0 0>;
+		link-hostwake-gpio = <&gpx1 1 0>;
+		link-active-gpio = <&gpf1 1 0>;
+		link-enable-gpio = <0>;
+
+		modem-type = "xmm6262";
+		link-type = "hsic";
+		modem-net = "umts";
+		use-handover = <0>;
+
+		pinctrl-names = "default", "active";
+		pinctrl-0 = <&modem_state_off>;
+		pinctrl-1 = <&modem_state_active>;
+	};
+
+};
+
+&pinctrl_1 {
+	modem_state_active: modem-state-active {
+		samsung,pins = "gpx1-6", "gpx1-1", "gpx1-2";
+		samsung,pin-pud = <0>;
+	};
+	modem_state_off: modem-state-off {
+		samsung,pins = "gpx1-6", "gpx1-1", "gpx1-2";
+		samsung,pin-pud = <1>;
+	};
 };
-- 
1.8.3.2

